addons/website_slides/models/slide_question.py

```diff
@@ -16,9 +16,14 @@ class SlideQuestion(models.Model):
     slide_id = fields.Many2one('slide.slide', string="Content", required=True)
     answer_ids = fields.One2many('slide.answer', 'question_id', string="Answer")
     # statistics
-    attempts_count = fields.Integer(compute='_compute_statistics', groups='website_slides.group_website_slides_officer')
+    attempts_count = fields.Integer(compute='_compute_statistics', groups='website_slides.group_website_slides_officer', store=True)
```

Comment:
Aye aye, this is the tricky point.

As you may have observed, attempts_count was not stored. This was done on purpose, to avoid too much stored fields at first implementation of quizzes.

I think here we should split the _compute_statistics methods in two. Indeed mixing stored and not stored fields in a compute is not a good idea, as not stored fields are computed at each transaction. This is not what we expect from a stored field as it would imply unnecessary write in DB.

For attempts_count itself, the issue is that it depends upon slide_id.slide_partner_ids (each slide - partner relationship will impact attempts_count on the question linked to the slide). This would be costly to compute in an automated computed field, and probably lead to concurrence access errors in a heavy duty scenario with lot of people taking quizzes at the same time. Computed fields have some computational cost.

My proposal is the following
  * make attempts_count stored as a classic field (not computed);
  * manually trigger its computation when creating a slide.slide.partner or updating its quiz_attempts_count field. You can have a look at _recompute_completion that does a bit the same kind of "manually computed field" (completed field on slide.channel.partner model). The underlying idea is to precisely compute a heavy-duty field to avoid concurrent accesses. It was proven to be really helpful on the completed field, I think you should try to mimic this implementation for attempts_count.

So what I propose
  * store attempts_count, not computed, update it manually;
  * keep compute methods for the other attempts fields, not stored. It will use the now-stored attempts_count field;
  * write unit tests to ensure your code is bug-free (in tests/ directory you have already existing test files you can add some of them for your feature);

Thanks !